src/shared/condition.c

```diff
@@ -578,8 +579,51 @@ static int condition_test_firmware_devicetree_compatible(const char *dtcarg) {
         return strv_contains(dtcompatlist, dtcarg);
 }
 
+static int condition_test_firmware_smbios_field(const char *expression) {
+        int r;
+        _cleanup_free_ char *field = NULL, *expected_value = NULL, *actual_value = NULL;
+        OrderOperator operator;
+
+        assert(expression);
+
+        /* Parse SMBIOS field */
+        r = extract_first_word(&expression, &field, "!<=>", EXTRACT_RETAIN_SEPARATORS);
+        if (r <= 0 || isempty(expression))
+                return -EINVAL;
+
+        /* Remove trailing spaces from SMBIOS field */
+        strstrip(field);
+
+        /* Parse operator */
+        operator = parse_order(&expression);
+        if (operator == -EINVAL)
+                return operator;
+
+        /* Parse expected value */
+        r = extract_first_word(&expression, &expected_value, NULL, EXTRACT_UNQUOTE);
+        if (r <= 0)
+                return -EINVAL;
+
+        /* Read actual value from sysfs */
+        const char *p = strjoina("/sys/class/dmi/id/", ascii_strlower(field));
```

Comment:
Typically we operate in case-sensitive fashion, hence I don't think we should do ascii_strlower()

Reply:
Also, please don't blindly append arbitrary strings to paths. Code hygiene demands you verify that the string is safe to append here. i.e. please check first if the string passes filename_is_valid(), and refuse things if not.

Reply:
Adjusted in latest version